Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -2014,22 +2014,29 @@ void MatrixWorkspace::rebuildDetectorIDGroupings() {
       const size_t timeIndex = index.second;
       if (detIndex >= allDetIDs.size()) {
         parallelException = true;
-        error = "MatrixWorkspace: SpectrumDefinition contains an out-of-range "
-                "detector index, i.e., the spectrum definition does not match "
-                "the instrument in the workspace.";
+        errorCode = 1;
       } else if (timeIndex >= detInfo.scanCount(detIndex)) {
         parallelException = true;
-        error = "MatrixWorkspace: SpectrumDefinition contains an out-of-range "
-                "time index for a detector, i.e., the spectrum definition does "
-                "not match the instrument in the workspace.";
+        errorCode = 2;
       } else {
         detIDs.insert(allDetIDs[detIndex]);
       }
     }
     spec.setDetectorIDs(std::move(detIDs));
   }
-  if (parallelException)
-    throw std::invalid_argument(error);
+  if (parallelException) {
```

Comment:
Can we remove `parallelException`? Check for non-zero `errorCode` enough?

Reply:
Good point - that should work.